.circleci/workflow-queue-v1.sh

```diff
@@ -54,22 +54,29 @@ circle_get() {
     "${url}"
 }
 
+# Function returns current pipeline's start_time. It is used for comparison of start_time values.
+fetch_current_pipeline_start_time() {
+  printf '%s\n' "Fetch current pipeline start time."
+  local get_path="project/${project_slug}/tree/${branch}?filter=running&shallow=true"
+  local curl_result=$(circle_get "${get_path}")
+  __=$(echo "${curl_result}" | jq -r ".[] | select(.build_num==$CIRCLE_BUILD_NUM) | .start_time")
+}
+
+# Function takes start_time parameter.
 # Fetch list of builds on master branch that are running, pending or queued.
-fetch_pipelines() {
-  printf '%s\n' "Fetch previously submitted builds on \"${branch}\" branch that are running, pending or queued:"
+fetch_older_pipelines() {
+  printf '%s\n' "Fetch pipelines (older than ${1}) on \"${branch}\" branch that are running, pending or queued."
   local get_path="project/${project_slug}/tree/${branch}?filter=running&shallow=true"
-  local get_result=$(circle_get "${get_path}")
-  if [[ ! "${get_result}" ]]; then
-    printf "curl failed."
+  local curl_result=$(circle_get "${get_path}")
+  if [[ ! "${curl_result}" ]]; then
+    printf "Fetch older pipelines failed."
     exit 1
   fi
-  jq_filter="select(.branch==\"${branch}\""
-  jq_filter+=" and .build_num < $CIRCLE_BUILD_NUM"
-  jq_filter+=" and (.status | test(\"running|pending|queued\")))"
-  jq_filter+=" | select(.workflows.workflow_name==\"${workflow_name}\""
-  jq_filter+=" and .workflows.workflow_id!=\"${CIRCLE_WORKFLOW_ID}\")"
-
-  __=$(echo "${get_result}" | jq -r ".[] | ${jq_filter} | .build_num | @sh")
+  jq_filter=".branch==\"${branch}\" and (.status | test(\"running|pending|queued\")) "
+  jq_filter+="and .workflows.workflow_name==\"${workflow_name}\" and .workflows.workflow_id!=\"${CIRCLE_WORKFLOW_ID}\""
+  jq_object="{ workflow_name: .workflows.workflow_name, workflow_id: .workflows.workflow_id, "
+  jq_object+="job_name: .workflows.job_name, build_num: .build_num, start_time: .start_time, status: .status }"
```

Comment:
Create a new JSON object for pretty print in CI console.

<img width="970" alt="Screen Shot 2021-08-13 at 1 24 29 PM" src="https://user-images.githubusercontent.com/35533885/129396876-98ecca2f-176b-43fb-bef8-e5ced03514bb.png">

Reply:
One simplification here for top-level fields:

```
..., build_num: .build_num, start_time: .start_time, status: .status}
```

-> 
```
..., build_num, start_time, status }"
```

Reply:
ah very cool. I didn't know about this. thank you.